proposals/eligibility_validator_test.go

```diff
@@ -0,0 +1,464 @@
+package proposals
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log/logtest"
+	"github.com/spacemeshos/go-spacemesh/signing"
+)
+
+var (
+	activeSet   = types.ATXIDList{types.RandomATXID(), types.RandomATXID()}
+	defaultUnit = uint(5)
+	ballotUnit  = uint(1)
+	// eligibleSlots is calculated based on layerAveSize, layersPerEpoch, epoch ATX weight and smesher's own weight.
+	eligibleSlots = uint32(3)
```

Comment:
uints can be constant, i generally don't see any reason to use global variables in tests and would be better to create initializers for common objects, e.g. genActiveSet(), genBeacon

Reply:
done